addons/stock/models/stock_picking.py

```diff
@@ -769,11 +769,28 @@ def _onchange_picking_type(self):
                     'message': partner.picking_warn_msg
                 }}
 
+    # This method is kept to not break stable by potential override in custom modules
+    # TODO: it should be removed in master
     @api.onchange('location_id', 'location_dest_id')
     def _onchange_locations(self):
+        pass
+
+    @api.onchange('location_id')
+    def _onchange_location_id(self):
         (self.move_ids | self.move_ids_without_package).update({
             "location_id": self.location_id,
-            "location_dest_id": self.location_dest_id
+        })
+        if any(line.reserved_qty or line.qty_done for line in self.move_ids.move_line_ids):
+            return {'warning': {
+                    'title': 'Locations to update',
```

Comment:
it was like this before your commit, but a little odd that the title is not translated

Reply:
Thanks for the review.

I've made it translatable to prevent a potential ticket about it :)